corehq/apps/hqwebapp/forms.py

```diff
@@ -48,22 +51,47 @@ def clean(self):
         if settings.ADD_CAPTCHA_FIELD_TO_FORMS:
             if not self.cleaned_data.get('captcha'):
                 raise ValidationError(_("Please enter valid CAPTCHA"))
-
-        try:
-            cleaned_data = super(EmailAuthenticationForm, self).clean()
-        except ValidationError:
-            user = CouchUser.get_by_username(username)
-            if user and user.is_locked_out():
-                metrics_counter('commcare.auth.lockouts')
-                raise ValidationError(LOCKOUT_MESSAGE)
-            else:
-                raise
         user = CouchUser.get_by_username(username)
         if user and user.is_locked_out():
             metrics_counter('commcare.auth.lockouts')
             raise ValidationError(LOCKOUT_MESSAGE)
+        try:
+            cleaned_data = super(EmailAuthenticationForm, self).clean()
+            if not user:
+                try:
+                    _ = ConsumerUser.objects.get(user=self.user_cache)
+                    EmailAuthenticationForm.create_web_user_from_consumer_user(username,
+                                                                               password,
+                                                                               get_ip(self.request))
+                except ConsumerUser.DoesNotExist:
+                    self.get_invalid_login_error()
+        except ValidationError:
+            raise
         return cleaned_data
 
+    @staticmethod
+    def create_web_user_from_consumer_user(username, password, ip=None, is_domain_admin=True):
+        now = datetime.utcnow()
+
+        new_user = WebUser.create(None, username, password, None, USER_CHANGE_VIA_WEB, is_admin=is_domain_admin)
+        new_user.subscribed_to_commcare_users = False
+        new_user.eula.signed = True
+        new_user.eula.date = now
+        new_user.eula.type = 'End User License Agreement'
+        if ip:
+            new_user.eula.user_ip = ip
+
+        new_user.is_staff = False  # Can't log in to admin site
```

Comment:
aren't a lot of these things already the defaults?

Reply:
Removed the fields which have defaults